cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -163,6 +176,15 @@ protected void startUp() throws Exception {
     }
 
     cancelHttpService = startHttpService(httpServiceBuilder.build());
+    TransactionRunners.run(transactionRunner, context -> {
+      AppMetadataStore appMetadataStore  = AppMetadataStore.create(context);
+      int count = 0;
+      for (NamespaceMeta namespaceMeta : namespaceQueryAdmin.list()) {
```

Comment:
The query admin list will happen in its own transaction. It shouldn't be called inside a transaction.

Reply:
removed namespace query as it's not required.